component/src/main/java/org/wso2/extension/siddhi/io/jms/sink/JMSSink.java

```diff
@@ -132,35 +133,25 @@ protected void init(StreamDefinition outputStreamDefinition, OptionHolder option
 
     @Override
     public void connect() throws ConnectionUnavailableException {
-        this.clientConnector = new JMSClientConnector();
-    }
-
-    @Override
-    public Map<String, Object> currentState() {
-        return null;
-    }
-
-    @Override
-    public void restoreState(Map<String, Object> state) {
-
+        try {
+            this.clientConnector = new JMSConnectorFactoryImpl().createClientConnector(jmsStaticProperties);
+        } catch (JMSConnectorException e) {
+            log.error("Error connecting to JMS provider. " + e.getMessage());
+        }
     }
 
     @Override
     public void publish(Object payload, DynamicOptions transportOptions) {
         String topicQueueName = destination.getValue(transportOptions);
         try {
-            executorService.submit(new JMSPublisher(topicQueueName, jmsStaticProperties, clientConnector, payload));
-        } catch (RejectedExecutionException e) {
-            //No need to retry as we are using an unbounded queue. Only place this can happen is when the executor
-            // service is shutting down. In such cases we can't anyway handle the exception. Hence logging.
-            log.error("Error occured when submitting following payload to be published via JMS. Payload : "
-                    + payload.toString(), e);
+            new JMSPublisher(topicQueueName, jmsStaticProperties, clientConnector, payload).run();
```

Comment:
We are creating one thread per each message here right? Wouldn't that be a problem?

Reply:
Fixed with commit [#3fb99a6822b8f5bb32d97ebdc4b7495aa01a0dfc](https://github.com/wso2-extensions/siddhi-io-jms/pull/25/commits/3fb99a6822b8f5bb32d97ebdc4b7495aa01a0dfc)